service/src/main/java/uk/gov/hmcts/reform/fpl/service/email/content/AbstractEmailContentProvider.java

```diff
@@ -24,14 +26,16 @@
 import static uk.gov.hmcts.reform.fpl.utils.EmailNotificationHelper.formatCaseUrl;
 import static uk.gov.hmcts.reform.fpl.utils.PeopleInCaseHelper.getFirstRespondentLastName;
 
-public abstract class AbstractEmailContentProvider {
-
+abstract class AbstractEmailContentProvider {
     final String uiBaseUrl;
-    private final HearingBookingService hearingBookingService;
 
-    protected AbstractEmailContentProvider(String uiBaseUrl, HearingBookingService hearingBookingService) {
+    @Autowired
+    ObjectMapper mapper;
+
+    private final HearingBookingService hearingBookingService = new HearingBookingService();
```

Comment:
Seeing as this is a service it could be autowired however the methods that use this are only used by `CafcassEmailContentProviderSDOIssued` and `LocalAuthorityEmailContentProvider`. Maybe it would be good to have an intermediary abstract class for those 2 that contains the `hearingBookingService` as well as the 2 methods that use it.

Reply:
Continuing on from this line of thought, there could be other abstract class' that add the functionality for the other methods in this class as not all the content providers need these methods.

Reply:
See [dev/more-content-provider-clean-up](https://github.com/hmcts/fpl-ccd-configuration/tree/dev/more-content-provider-clean-up) for a possibility